### PR TITLE
Apollo Local State managment usage (The Return)

### PR DESCRIPTION
**Pull request by estacioneto**
**Related to vtex-apps/minicart#115**

#### What is the purpose of this pull request?

- Use the Apollo Local State API and delete the store-resources dependency

#### What problem is this solving?

We want to remove the orderFormContext and to do so, we need to put the orderForm into the local state.

#### How should this be manually tested?

[Access the workspace](https://offline--storecomponents.myvtex.com/)

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/BuyButton/index.js [hunk 1]
```diff
@@ -44,47 +40,75 @@ export class BuyButton extends Component {
...
+          skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
         )
 
-        orderFormContext.refetch()
-        this.toastMessage(success.length >= 1)
-      },
-      () => {
-        this.toastMessage(false)
-      }
+      if (isOneClickBuy) location.assign(CONSTANTS.CHECKOUT_URL)
```
Comment (luciannojunior) [inline react/components/BuyButton/index.js]: Check ident

File: react/components/Greeting/index.js [hunk 2]
```diff
@@ -18,8 +38,9 @@ const withWrapper = Component => props => (
   </Wrapper>
 )
 
-const Greeting = ({ orderFormContext }) => {
-  const firstName = path(['orderForm', 'clientProfileData', 'firstName'], orderFormContext)
+const Greeting = ({ orderForm }) => {
+  console.log('orderForm', orderForm)
```
Comment (luciannojunior) [inline react/components/Greeting/index.js]: Remove

File: react/testUtils/intl-helper.js [hunk 3]
```diff
@@ -2,7 +2,6 @@ import React from 'react'
 import * as reactTestingLibrary from 'react-testing-library'
 import { IntlProvider } from 'react-intl'
 import defaultStrings from '../../messages/en-US.json'
-const Context = React.createContext({})
```
Comment (luciannojunior) [inline react/testUtils/intl-helper.js]: This was unused?

File: react/__tests__/components/Greeting.test.js [hunk 4]
```diff
@@ -1,18 +1,36 @@
...
 import React from 'react'
+import { MockedProvider } from 'react-apollo/test-utils'
 import { render } from 'intl-helper'
 
 import Greeting from '../../Greeting'
 
 describe('<Greeting /> component', () => {
+  const data = {
+    minicart: {
+      __typename: 'Minicart',
```
[the remark this answers is not in the document]
Comment (estacioneto) [reply react/__tests__/components/Greeting.test.js]: Yes. This is related to the local state cache API. To map it correctly

**Pull request merged**